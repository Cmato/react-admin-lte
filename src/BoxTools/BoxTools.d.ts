import * as React from 'react';

export interface BoxToolsProps {
}

declare const BoxTools: React.ComponentType<BoxToolsProps>;

export default BoxTools;
