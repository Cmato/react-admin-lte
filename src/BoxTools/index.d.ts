export { default } from './BoxTools';
export * from './BoxTools';
