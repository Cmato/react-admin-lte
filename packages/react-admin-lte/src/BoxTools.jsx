import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

const propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

const BoxTools = ({ className, children }) => {
  const classes = {
    'box-tools': true,
  };

  return (
    <div className={classNames(className, classes)}>
      {children}
    </div>
  );
};

BoxTools.propTypes = propTypes;

export default BoxTools;
