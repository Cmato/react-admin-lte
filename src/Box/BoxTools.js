import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const BoxTools = ({
  children,
  className: classNameProp,
  ...other
}) => {
  const className = classNames({
    'box-tools': true,
    'pull-right': true
  }, classNameProp);

  return (
    <div className={className} {...other}>
      {children}
    </div>
  );
};

BoxTools.propTypes = {
  /**
   * The contents of the box tools.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string
};

export default BoxTools;
