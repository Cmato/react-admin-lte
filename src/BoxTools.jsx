import React from 'react';
import classNames from 'classnames';

const propTypes = {
  children: React.PropTypes.node,
};

const BoxTools = ({ className, children }) => {
  const classes = {
    'box-tools': true,
    'pull-right': true,
  };

  return (
    <div className={classNames(className, classes)}>
      {children}
    </div>
  );
};

BoxTools.propTypes = propTypes;

export default BoxTools;
