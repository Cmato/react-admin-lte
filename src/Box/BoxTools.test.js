import React from 'react';
import { shallow } from 'enzyme';
import BoxTools from './BoxTools';

it('renders default', () => {
  expect(shallow(<BoxTools><div /></BoxTools>))
    .toMatchSnapshot();
});

