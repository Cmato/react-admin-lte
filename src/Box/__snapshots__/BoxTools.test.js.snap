// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default 1`] = `
<div
  className="box-tools pull-right"
>
  <div />
</div>
`;
